src/components/tasks/card/index.tsx

```diff
@@ -257,25 +265,148 @@ const Card: FC<Props> = ({
     const onEditEnabled = () => {
         actions.onEditRoute();
     };
+    const EditButton = () => (
+        <div className={classNames.editButton} data-testid="edit-button">
+            <Image
+                src="/pencil.webp"
+                alt="edit"
+                width={iconWidth}
+                height={iconHeight}
+                onClick={onEditEnabled}
+                tabIndex={0}
+            />
+        </div>
+    );
+
+    const ProgressIndicator = () => (
+        <div className={classNames.progressIndicator}>
+            <div
+                className={`
+                ${handleProgressColor(
+                    content.percentCompleted,
+                    content.startedOn,
+                    content.endsOn
+                )}
+                ${classNames.progressStyle}
+                `}
+                style={{
+                    width: `${content.percentCompleted}%`,
+                }}
+            ></div>
+        </div>
+    );
+    const CardTitle = () => (
+        <h2
+            className={classNames.cardTitle}
+            contentEditable={shouldEdit}
+            onKeyPress={(e) => handleChange(e, 'title')}
+            role="button"
+            tabIndex={0}
+        >
+            {cardDetails.title}
+        </h2>
+    );
+
+    // show redesign only on dev
+    if (isNewCardEnabled)
```

Comment:
this piece of code needs to be refactored

Reply:
will be tracking this with Issue #497